homeassistant/components/weather/__init__.py

```diff
@@ -192,15 +221,28 @@ def state_attributes(self):
             data[ATTR_WEATHER_OZONE] = ozone
 
         if (pressure := self.pressure) is not None:
+            if (unit := self.pressure_unit) is not None:
+                pressure = round(
+                    self.hass.config.units.pressure(pressure, unit), ROUNDING_PRECISION
+                )
             data[ATTR_WEATHER_PRESSURE] = pressure
 
         if (wind_bearing := self.wind_bearing) is not None:
             data[ATTR_WEATHER_WIND_BEARING] = wind_bearing
 
         if (wind_speed := self.wind_speed) is not None:
+            if (unit := self.wind_speed_unit) is not None:
+                wind_speed = round(
+                    self.hass.config.units.wind_speed(wind_speed, unit),
```

Comment:
Why is hass config doing the unit conversion?

`wind_seed` should be imported directly from `util.unit_system` instead.

The same applies to all other unit conversions happening here.